asteroid_filterbanks/enc_dec.py

```diff
@@ -130,15 +131,26 @@ class Encoder(_EncDec):
             If True (default), forwarding input with shape (batch, 1, time)
             will output a tensor of shape (batch, freq, conv_time).
             If False, will output a tensor of shape (batch, 1, freq, conv_time).
-        padding (int): Zero-padding added to both sides of the input.
+        padding (Union[int, List[int]): Zero-padding added to both sides of the input.
+        pad_mode (str): See torch.nn.functional.pad.
 
     """
 
-    def __init__(self, filterbank, is_pinv=False, as_conv1d=True, padding=0):
+    def __init__(
+        self, filterbank, is_pinv=False, as_conv1d=True, center=False, padding=0, pad_mode="reflect"
+    ):
         super(Encoder, self).__init__(filterbank, is_pinv=is_pinv)
         self.as_conv1d = as_conv1d
+        self.center = center
+        if padding and center:
+            warnings.warn(f"Found `center=True` and `padding={padding}`. Padding takes precedence.")
+            self.padding = padding
+        elif center:
+            self.padding = [self.filterbank.kernel_size // 2, self.filterbank.kernel_size // 2]
+        else:
+            self.padding = padding
+        self.pad_mode = pad_mode
```

Comment:
This is not very pretty but I don't know what else would make sense.

Reply:
Does this work for uneven kernel sizes ( e.g. center=True for free filterbank ) ?

Reply:
Maybe not, I'd have to check. Also that's why I don't like the `center` everywhere option..